estuary_updater/handlers/base.py

```diff
@@ -48,3 +54,83 @@ def handle(self, msg):
         :param dict msg: a message to be processed
         """
         pass
+
+    @property
+    def koji_session(self):
+        """
+        Get a cached Koji session but initialize the connection first if needed.
+
+        :return: a Koji session object
+        :rtype: koji.ClientSession
+        """
+        if not self._koji_session:
+            self._koji_session = koji.ClientSession(self.config['estuary_updater.koji_url'])
+        return self._koji_session
+
+    def is_container_build(self, build_info):
+        """
+        Check whether a Koji build is a container build.
+
+        :param KojiBuild build_info: build info from the Koji API
+        :return: boolean value indicating whether the build is a container build
+        :rtype: bool
+        """
+        package_name = build_info['package_name']
+        # Checking a heuristic for determining if a build is a container build since, currently
+        # there is no definitive way to do it.
+        if build_info['extra'] and (
+                build_info['extra'].get('container_koji_build_id') or
+                build_info['extra'].get('container_koji_task_id')):
+            return True
+        # Checking another heuristic for determining if a build is a container build since
+        # currently there is no definitive way to do it.
+        elif (package_name.endswith('-container') or package_name.endswith('-docker')):
+            return True
+        else:
+            return False
+
+    def get_or_create_build(self, identifier, original_nvr=None, force_container_label=False):
+        """
+        Get a Koji build from Neo4j, or create it if it does not exist in Neo4j.
+
+        :param str/int identifier: an NVR (str) or build ID (int)
+        :kwarg str original_nvr: original_nvr property for the ContainerKojiBuild
+        :kwarg bool force_container_label: skips the check to see if the build is a container and
```

Comment:
maybe say that this happens when it's set to True, just to clarify. This could sound like it does the skip when it has its default value (False)